lib/SPIRV/SPIRVWriter.cpp

```diff
@@ -4844,6 +4844,17 @@ bool LLVMToSPIRVBase::transExecutionMode() {
       case spv::ExecutionModeNumSIMDWorkitemsINTEL:
       case spv::ExecutionModeSchedulerTargetFmaxMhzINTEL:
       case spv::ExecutionModeMaxWorkDimINTEL:
+      case spv::internal::ExecutionModeRegisterMapInterfaceINTEL: {
+        if (!BM->isAllowedToUseExtension(
+                ExtensionID::SPV_INTEL_kernel_attributes))
+          break;
+        AddSingleArgExecutionMode(static_cast<ExecutionMode>(EMode));
+        BM->addExtension(ExtensionID::SPV_INTEL_kernel_attributes);
+        // CapabilityFPGAKernelAttributesINTELv2 implicitly defines
+        // CapabilityFPGAKernelAttributesINTEL
+        BM->addCapability(CapabilityFPGAKernelAttributesINTEL);
+        BM->addCapability(CapabilityFPGAKernelAttributesINTELv2);
```

Comment:
AFAIK the translator isn't capable to track implicitly declared capabilities, it does look like a good enhancement for us. Like you did it here is LGTM